core/src/main/java/org/elasticsearch/cluster/metadata/IndexMetaData.java

```diff
@@ -973,20 +973,26 @@ public IndexMetaData build() {
                     filledInSyncAllocationIds.put(i, Collections.emptySet());
                 }
             }
-            final Map<String, String> requireMap = INDEX_ROUTING_REQUIRE_GROUP_SETTING.get(settings).getAsMap();
+            final Settings requireGroupSettings = INDEX_ROUTING_REQUIRE_GROUP_SETTING.get(settings);
+            DiscoveryNodeFilters.IP_VALIDATOR.accept(requireGroupSettings);
```

Comment:
thanks for alerting me to this, i didn't realize a validator could be set on group settings until you mentioned it.  I fixed this and all other occurrences below